cmd/server.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/metal-toolbox/conditionorc/internal/app"
+	"github.com/metal-toolbox/conditionorc/internal/model"
+	"github.com/metal-toolbox/conditionorc/internal/server"
+	"github.com/metal-toolbox/conditionorc/internal/store"
+	"github.com/spf13/cobra"
+	"go.hollow.sh/toolbox/events"
+)
+
+var (
+	shutdownTimeout = 10 * time.Second
+)
+
+// install server command
+var cmdServer = &cobra.Command{
+	Use:   "server",
+	Short: "Run condition orchestrator API service",
+	Run: func(cmd *cobra.Command, args []string) {
+		app, termCh, err := app.New(cmd.Context(), model.AppKindServer, cfgFile, model.LogLevel(logLevel))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		repository, err := store.NewStore(cmd.Context(), app.Config, app.Config.ConditionDefinitions, app.Logger)
+		if err != nil {
+			app.Logger.Fatal(err)
+		}
+
+		streamBroker, err := events.NewStream(app.Config.NatsOptions)
+		if err != nil {
+			app.Logger.Fatal(err)
+		}
+
+		if err := streamBroker.Open(); err != nil {
+			app.Logger.Fatal(err)
+		}
+
+		options := []server.Option{
+			server.WithLogger(app.Logger),
+			server.WithListenAddress(app.Config.ListenAddress),
+			server.WithStore(repository),
+			server.WithStreamBroker(streamBroker),
+			server.WithConditionDefinitions(app.Config.ConditionDefinitions),
+		}
+
+		srv := server.New(options...)
+		go func() {
+			if err := srv.ListenAndServe(); err != nil && errors.Is(err, http.ErrServerClosed) {
+				app.Logger.Fatal(err)
+			}
+		}()
+
+		// sit around for term signal
+		<-termCh
+		app.Logger.Info("got TERM signal, shutting down server...")
+
+		// call server shutdown with timeout
+		ctx, cancel := context.WithTimeout(cmd.Context(), shutdownTimeout)
+		defer cancel()
+		if err := srv.Shutdown(ctx); err != nil {
+			app.Logger.Fatal("server shutdown error:", err)
+		}
+
+		// wait until context is done
+		<-ctx.Done()
```

Comment:
`srv.Shutdown` is being passed a context with a deadline [here](https://github.com/metal-toolbox/conditionorc/pull/1/files/d1a43d342f546a3a0077cd2aee0dc67e7ee89de4#diff-ee2560d627cd77f1563972217a33ac39a4bf56fbf30d7f560fbb4199b9f70197R66), unless you think this may not be enough?

Reply:
Right, I wasn't very clear (and I misread your code 😅 ). Once the `Shutdown` call is complete, there isn't a need to wait on that context to timeout any more, so we can skip lines 73-77 IMO.